### DISCUSSION TITLE
📣 Contribute to the 2022 Web Almanac

### DISCUSSION THREAD

**Discussion by rviscomi** [HTTPArchive/almanac.httparchive.org#2906] [General]
Hi @HTTPArchive/almanac-contributors, please join us for the 2022 Web Almanac, our fourth edition!

At this early stage, we're starting with assigning teams of contributors to each chapter. A chapter's content team consists of [authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and [editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide). See [this discussion](https://github.com/HTTPArchive/almanac.httparchive.org/discussions/2165) for more info about how authors are selected.

ℹ️  **To get involved, choose a chapter from the list below and reply to its respective issue to indicate which role(s) you'd be interested in.** You can also filter issues for those labelled ["2022 chapter" and "help wanted"](https://github.com/HTTPArchive/almanac.httparchive.org/issues?q=is%3Aopen+is%3Aissue+label%3A%222022+chapter%22+label%3A%22help+wanted%22) to see what's available.

Here's the current lineup for 2022:

#### Page Content

- [CSS](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2879)
- [JavaScript](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2880)
- [Markup](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2881)
- [Structured Data](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2882)
- [Fonts](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2883)
- [Media](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2884)
- [WebAssembly](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2885)
- [Third Parties](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2886)
- [Interoperability 🆕](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2887)

#### User Experience

- [SEO](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2888)
- [Accessibility](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2889)
- [Performance](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2890)
- [Privacy](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2891)
- [Security](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2892)
- [Mobile Web](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2893)
- [Capabilities](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2894)
- [PWA](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2895)

#### Content Publishing

- [CMS](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2896)
- [Ecommerce](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2897)
- [Jamstack](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2898)
- [Sustainability 🆕](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2910)

#### Content Distribution

- [Page Weight](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2899)
- ~Resource Hints~ now under [Performance](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2890)
- [CDN](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2900)
- [Compression](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2901)
- [Caching](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2902)
- [HTTP](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2903)

⚠️ **Are there any other major topics on the web that we're missing?** Any suggestions to organize the content differently? I'd love to hear your ideas in the comments.

For all chapter-specific questions/discussions, head over to the issues linked above.

And if you're interested to contribute to the project beyond a specific chapter, there are also opportunities to join the [translators](https://github.com/HTTPArchive/almanac.httparchive.org/issues/923), [developers](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2904), and [designers](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2905) teams!

The Web Almanac is supported entirely by the web community, so **thank you all** for your interest and support to keep this valuable resource going year after year!

----

### _If you know which chapter you'd like to join, no need to reply to this discussion. Head over to the chapter's tracking issue and leave a comment indicating what role you're interested in!_

**Comment by daKmoR** [comment]
would it make sense to have a dedicated section for web components? or where would they fall into?

**Reply by rviscomi** [reply]
Good question. [Searching](https://almanac.httparchive.org/en/search?q=%22web+components%22#gsc.tab=0&gsc.q=%22web%20components%22&gsc.page=1) the 2021 edition shows it discussed in the JavaScript, Markup, and Page Weight chapters. I think the best fit might be the Markup chapter.

Do you feel like there's a sufficient amount of data to explore for a dedicated Web Components chapter?

**Reply by daKmoR** [reply]
things I can think of from the top of my head
- websites having at least one customElement defined
- average numbers of customElements if (at least one is defined)
- average javascript size/loading time of websites using at least one customElement
- how many websites are using shadow dom
- how many websites are using declarative shadow dom
- how many websites are using constructible stylesheets
- which web component libraries are used the most
- ...

hmmm I guess there could be enough content - what do you think?

PS: I am however not sure if these numbers are available? 🤔

**Reply by rviscomi** [reply]
I think there's some good data there but it might not be substantive enough to support its own chapter. The [2021 Markup](https://almanac.httparchive.org/en/2021/markup) chapter had about 35 metrics, for comparison. I'd suggest merging this with Markup for now and we can revisit this later if we find that there's sufficient data.

**Reply by GregBrimble** [reply]
Does the Jamstack chapter plan to include discussion about some of the newer web frameworks which start to blur the lines (e.g. Remix, SvelteKit, Qwik and Astro)?

**Reply by bkardell** [reply]
@rviscomi in markup in 2019 we started counting all 'non-standard elements' and using tools that used that collected data to classify 'dasherized' elements, but I don't think that then or since we have actually attempted to collect actual calls to `define`. That could be interesting because I am sure there is quite a mismatch between "I'm just using a safe element name I can find and style" vs "this is actually a custom element".  shadow roots would be interesting to know but unless the client has some special privileges, is it fully detectable? Just looking for `.shadowRoot` on everything won't work, will it? I think even that is not exposed if it is a closed root.

**Reply by rviscomi** [reply]
@bkardell that's a great question. I don't know enough about how custom elements are defined to say how feasible it'd be to detect but let's continue exploring that in the Markup chapter.

**Comment by rviscomi** [comment]
(forking to a new comment thread) @GregBrimble said:

> Does the Jamstack chapter plan to include discussion about some of the newer web frameworks which start to blur the lines (e.g. Remix, SvelteKit, Qwik and Astro)?

Nothing has been planned yet. A quick check of [Wappalyzer](https://www.wappalyzer.com/technologies/) shows that all of the example frameworks are indeed being detected already, which is great. I'd suggest accessing the [Jamstack 2022 planning doc](https://docs.google.com/document/d/15RLaaTVqoqb5AuDrlBt6L0J_BMx1ltKW4t8VWX-sN_g/edit?usp=sharing) and adding any ideas to the outline section. As the team fills out you can continue iterating on the outline.

I've also put your name down as a reviewer for now in the Jamstack issue #2898.

**Comment by SaptakS** [comment]
I wonder if there is scope for a chapter on [Indieweb](https://indieweb.org/). I am not sure it will have enough data tobe it's own chapter, but might be interesting to see how many websites are integrating indieweb features such as microformats, webmentions, etc.

**Reply by nhoizey** [reply]
This would be really great indeed! 😍

**Reply by rviscomi** [reply]
Perhaps we could incubate this topic in the [Structured Data](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2882) chapter and split it out if it turns out that there's enough content?
